File: admin/src/main/java/com/beat/admin/application/dto/result/AdminPromotionResults.java
package com.beat.admin.application.dto.result;

import java.util.List;

public record AdminPromotionResults(
	List<AdminPromotionResult> promotionResults
) {

	public AdminPromotionResults {
		promotionResults = List.copyOf(promotionResults);
	}

	public static AdminPromotionResults from(List<AdminPromotionResult> promotionResults) {
		return new AdminPromotionResults(promotionResults);
	}

	public record AdminPromotionResult(
		Long promotionId,
		String carouselNumber,
		String newImageUrl,
		boolean isExternal,
		String redirectUrl,
		Long performanceId
	) {
		public static AdminPromotionResult of(Long promotionId, String carouselNumber, String newImageUrl,
			boolean isExternal, String redirectUrl, Long performanceId) {
			return new AdminPromotionResult(promotionId, carouselNumber, newImageUrl, isExternal, redirectUrl,
				performanceId);
		}
	}
}

File: application/admin/src/main/kotlin/com/beat/application/admin/promotion/AdminPromotionResults.kt
package com.beat.application.admin.promotion

data class AdminPromotionResults(val promotionResults: List<AdminPromotionResult>) {
    data class AdminPromotionResult(
        val promotionId: Long,
        val carouselNumber: String,
        val newImageUrl: String,
        val isExternal: Boolean,
        val redirectUrl: String,
        val performanceId: Long?,
    )
}
